### PR TITLE
*Adding a MASK_OUTSIDE_OBCS option.

### PR DESCRIPTION
**Pull request by kshedstrom**
Adding a MASK_OUTSIDE_OBCS option.

- Use to mask out the regions inside the computational domain, but outside of all OBCs.
- Also changing the geometry of rotated Kelvin wave cases, changing answers.

### CODE REVIEW

**Comment by adcroft** [comment]
I'll take a closer look tomorrow but in the meantime static builds are failing with:
```
build/gnu/repro/static/ocean_only/double_gyre/MOM6: ../../../../../../..//src/core/MOM_open_boundary.F90:2368:36:
build/gnu/repro/static/ocean_only/double_gyre/MOM6: 
build/gnu/repro/static/ocean_only/double_gyre/MOM6:    integer ALLOCABLE_, dimension(:,:)  :: color     ! For sorting inside from outside
build/gnu/repro/static/ocean_only/double_gyre/MOM6:                                     1
build/gnu/repro/static/ocean_only/double_gyre/MOM6: Error: Array 'color' at (1) cannot have a deferred shape
build/gnu/repro/static/ocean_only/double_gyre/MOM6: ../../../../../../..//src/core/MOM_open_boundary.F90:2369:36:
build/gnu/repro/static/ocean_only/double_gyre/MOM6: 
build/gnu/repro/static/ocean_only/double_gyre/MOM6:    real ALLOCABLE_, dimension(:,:)     :: depth     ! Global T-point land mask
build/gnu/repro/static/ocean_only/double_gyre/MOM6:                                     1
build/gnu/repro/static/ocean_only/double_gyre/MOM6: Error: Array 'depth' at (1) cannot have a deferred shape
build/gnu/repro/static/ocean_only/double_gyre/MOM6: ../../../../../../..//src/core/MOM_open_boundary.F90:2376:11:
build/gnu/repro/static/ocean_only/double_gyre/MOM6: 
build/gnu/repro/static/ocean_only/double_gyre/MOM6:    allocate(color(G%isg-1:G%ieg+1, G%jsg-1:G%jeg+1)) ; color = 0
build/gnu/repro/static/ocean_only/double_gyre/MOM6:            1
build/gnu/repro/static/ocean_only/double_gyre/MOM6: Error: Allocate-object at (1) must be ALLOCATABLE or a POINTER
build/gnu/repro/static/ocean_only/double_gyre/MOM6: ../../../../../../..//src/core/MOM_open_boundary.F90:2377:11:
build/gnu/repro/static/ocean_only/double_gyre/MOM6: 
build/gnu/repro/static/ocean_only/double_gyre/MOM6:    allocate(depth(G%isg:G%ieg, G%jsg:G%jeg))         ; depth = 0
build/gnu/repro/static/ocean_only/double_gyre/MOM6:            1
build/gnu/repro/static/ocean_only/double_gyre/MOM6: Error: Allocate-object at (1) must be ALLOCATABLE or a POINTER
build/gnu/repro/static/ocean_only/double_gyre/MOM6: ../../../../../../..//src/core/MOM_open_boundary.F90:2399:13:
build/gnu/repro/static/ocean_only/double_gyre/MOM6: 
build/gnu/repro/static/ocean_only/double_gyre/MOM6:    deallocate(depth)
build/gnu/repro/static/ocean_only/double_gyre/MOM6:              1
build/gnu/repro/static/ocean_only/double_gyre/MOM6: Error: Allocate-object at (1) must be ALLOCATABLE or a POINTER
build/gnu/repro/static/ocean_only/double_gyre/MOM6: ../../../../../../..//src/core/MOM_open_boundary.F90:2445:13:
build/gnu/repro/static/ocean_only/double_gyre/MOM6: 
build/gnu/repro/static/ocean_only/double_gyre/MOM6:    deallocate(color)
build/gnu/repro/static/ocean_only/double_gyre/MOM6:              1
build/gnu/repro/static/ocean_only/double_gyre/MOM6: Error: Allocate-object at (1) must be ALLOCATABLE or a POINTER
build/gnu/repro/static/ocean_only/double_gyre/MOM6: make[1]: *** [MOM_open_boundary.o] Error 1
build/gnu/repro/static/ocean_only/double_gyre/MOM6: make[1]: *** Waiting for unfinished jobs....
MRS/Makefile.build:197: recipe for target 'build/gnu/repro/static/ocean_only/double_gyre/MOM6' failed
make: *** [build/gnu/repro/static/ocean_only/double_gyre/MOM6] Error 11
```

File: .travis.yml [hunk 1]
```diff
@@ -41,7 +41,7 @@ script:
  - ./house_cleaning/trailer.py -e TEOS10 src config_src
  - bash MOM6-examples/tools/tests/Travis-MOM6/build_ocean_only.sh
  - bash MOM6-examples/tools/tests/Travis-MOM6/build_symmetric_ocean_only.sh
- - travis_wait bash MOM6-examples/tools/tests/Travis-MOM6/run_tests.sh
+ - bash MOM6-examples/tools/tests/Travis-MOM6/run_tests.sh
```
Comment (adcroft) [inline .travis.yml]: I believe I have solved the timeout problem. See NOAA-GFDL/MOM6-examples@c95392e2f203e5e54 .

**Comment by kshedstrom** [comment]
For the static memory, I know how big those arrays should be:
> allocate(color(G%isg-1:G%ieg+1, G%jsg-1:G%jeg+1)) ; color = 0
There's nothing in MOM_memory_macros.h for global arrays. That's what we need, right?

**Comment by kshedstrom** [comment]
I think I fixed it - without messing with MOM_memory_macros.h. Try again?

**Comment by adcroft** [comment]
> For the static memory, I know how big those arrays should be:
>   allocate(color(G%isg-1:G%ieg+1, G%jsg-1:G%jeg+1)) ; color = 0
> There's nothing in MOM_memory_macros.h for global arrays. That's what we
need, right?

I don't think you need global arrays at all for this algorithm - Why can't
you use the usual arrays with a halo and update at each iteration of the
flood?

--
Dr Alistair Adcroft    (Alistair.Adcroft@noaa.gov)
Princeton University           Tel: (609) 987-5073
NOAA/GFDL, 201 Forrestal Road, Princeton, NJ 08540

**Comment by kshedstrom** [comment]
That's another way to do it. It hadn't occurred to me, but I'm not sure I
can do it correctly with the end state check communications.

Kate

On Thu, Oct 26, 2017 at 12:39 PM, Alistair Adcroft <notifications@github.com

**Comment by adcroft** [comment]
Am I right in thinking you are still re-working this particular PR? Shall we close it until you are done?

**Comment by kshedstrom** [comment]
If you're asking about the MASK_OUTSIDE stuff, I believe that is working
correctly. Other updates will be coming, related to other issues.

**Comment by kshedstrom** [comment]
I fixed the merge conflict and the difference with PE counts in ESMG seamount. Go ahead on the PR?

**Pull request merged**